### ISSUE TITLE
Analyze Chrome browser extensions (CRX)

### ISSUE THREAD

**Issue by pdelsante** [intelowlproject/IntelOwl#71]
Browser extensions in Chrome are identified by an alphanumeric identifier.

For example, [Google Keep](https://chrome.google.com/webstore/detail/google-keep-chrome-extens/lpcaedmchfhocbbapmcbpinfpgnhiddi) has the following URL:

```
https://chrome.google.com/webstore/detail/google-keep-chrome-extens/lpcaedmchfhocbbapmcbpinfpgnhiddi
```

which leads to the extension ID `lpcaedmchfhocbbapmcbpinfpgnhiddi`.

Extension IDs are used to determine the installation folder, e.g.

```
C:\Users\[login_name]\AppData\Local\Google\Chrome\User Data\Default\Extensions\lpcaedmchfhocbbapmcbpinfpgnhiddi
```

Also, in normal mode (not in incognito windows) local resources injected by an extension inside web pages is loaded by using urls of this form:

```
chrome-extension://lpcaedmchfhocbbapmcbpinfpgnhiddi/resource-name.js
```

Knowing whether an extension is malicious or not is a very useful task in many cases, and several public services/APIs already exist for this purpose, for example:
- https://crxcavator.io/
- https://robwu.nl/crxviewer/

Being able to analyze CRX IDs would be very useful IMHO.

**Comment by codernishchay** [comment]
I want to work on this issue. Please assign it to me?

**Comment by mlodic** [comment]
Hi @codernishchay, thank you for your interest in the project! You have been assigned!

Even if it was flagged as "beginner-friendly", this issue could tricky to be correctly implemented.

This analyzer is particular if compared with others because it analyzes a different kind of observable that is actually not managed: a Chrome extension ID. 

This means that some changes in the client applications are also required to let the new analyzer work:
* change this [function](https://github.com/intelowlproject/pyintelowl/blob/7c03f8325d58be9eaa3587f4c0476c0b9fe0c460/pyintelowl/pyintelowl.py#L171) in pyintelowl client to support Chrome extension ID observable type.
* another change in IntelOwl-ng to let the Angular app accept the Chrome extension ID. @Eshaan7 could you please help us / provide opinion on this?

Ultimately I obviously suggest you to follow the steps in the documentation and ask for help if needed!

**Comment by Eshaan7** [comment]
@mlodic You are right. This would require changes in client. I think we discussed this before, right ? We could add a type like `IoC` or `jolly` which would accept any kind of string and we could add this same name to some analyzers' `observable_supported`. This `IoC` could then work for email, chrome id, username, etc. For example, I think `IntelX_Search` works for email and usernames which we do not yet support so could be a nice addition.

**Comment by mlodic** [comment]
I agree. In this way we can manage all the other observable types (the most rare ones) at once.

**Comment by CITIZENDOT** [comment]
Hey, I would like to work on this, if this isn't active. Currently we have generic type in observable and so, It could be easy to implement I suppose.

**Comment by mlodic** [comment]
Thank you, you are assigned!

Yes, I would use the `generic` type.

About creating a single analyzer or not, we usually separate them but this is a particular analysis type so it could be helpful to keep them together to help the ones that need to look for this type of analysis. Please try and see how it comes up

**Comment by mlodic** [comment]
merged #318
